src/app/navigation/routines-stack.tsx

```diff
@@ -0,0 +1,94 @@
+import { ArrowLeft } from 'lucide-react-native';
+import { useTheme } from 'styled-components';
+import React, { FC, useMemo } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { BottomTabScreenProps } from '@react-navigation/bottom-tabs';
+import { createStackNavigator } from '@react-navigation/stack';
+
+import {
+  AddRoutine,
+  ExercisesRevision,
+  RoutineDashboard,
+  SelectRoutineExercises,
+} from 'src/app/screens/routines';
+import { Exercise } from 'src/interfaces/exercises';
+import mockedExercises from 'src/mocks/weekly-exercises-data.json';
+
+import { Header } from '../components';
+import { createFormConfig, FormData } from '../screens/routines/form-config';
+import { BottomTabParamList, RoutinesParamList } from './types';
+
+const Stack = createStackNavigator<RoutinesParamList>();
+
+interface Props extends BottomTabScreenProps<BottomTabParamList> {}
+
+export const RoutineStack: FC<Props> = ({ navigation }) => {
+  const { t } = useTranslation();
+  const theme = useTheme();
+
+  const formConfig = createFormConfig(mockedExercises as Exercise[]);
+  const methods = useForm<FormData>(formConfig);
+
+  const goBackHeaderOptions = useMemo(
+    () => ({
+      header: () => (
+        <Header
+          headerLeft={
+            <ArrowLeft
+              color={theme.colors.background}
+              onPress={() => {
+                navigation.goBack();
+              }}
+            />
+          }
+        />
+      ),
+    }),
+    [navigation, theme],
+  );
```

Comment:
If you want something more performant than this, you could create this function outside the component and add the params navigation and theme. Then you call the function using the params.

WDYT?

Or can you review how we do it on the other stacks?